core/src/main/scala/org/http4s/headers/Age.scala

```diff
@@ -1,5 +1,31 @@
 package org.http4s
 package headers
 
-object Age extends HeaderKey.Default
+import org.http4s.util.{Renderer, Writer}
+import org.http4s.util.Renderable._
+import org.http4s.parser.HttpHeaderParser
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+object Age extends HeaderKey.Internal[Age] with HeaderKey.Singleton {
+  override def parse(s: String): ParseResult[Age] =
+    HttpHeaderParser.AGE(s)
+}
+
+/**
+  * Constructs an Age header.
+  *
+  * The value of this field is a positive number of seconds (in decimal) with an estimate of the amount of time since the response
+  *
+  * @param age age of the response
+  */
+final case class Age(age: FiniteDuration) extends Header.Parsed {
+  require(age >= 0.seconds)
```

Comment:
In the `SimpleHeadersParser` would it be ok to use `unsafe`?

```
      Digits ~ EOL ~> ((t: String) => Age.unsafe(FiniteDuration(t.toLong, SECONDS)))
```

We know from the parser that t must be positive

Or is there a better solution considering that `Age.apply` will return `ParseResult`?